lib/jupyter/actions.js

```diff
@@ -381,6 +381,34 @@ define([
         );
       }
     }, 'extend-selection-below', 'vim-binding');
+
+    var cell_marker = {};
+
+    actions.register({
+      'handler': function(env, event) {
+        env.notebook.keyboard_manager.enabled = false;
+        document.addEventListener('keydown', function (event) {
+          var key = event.key;
+          var idx = env.notebook.get_selected_index();
+          cell_marker[key] = idx;
+          env.notebook.keyboard_manager.enabled = true;
```

Comment:
Should enable `keyboard_manager` first so that any exceptions in `env.notebook.get_selected_index()` won't break things

```suggestion
          env.notebook.keyboard_manager.enabled = true;
          var key = event.key;
          var idx = env.notebook.get_selected_index();
          cell_marker[key] = idx;
```